navigation/src/main/java/com/graphhopper/navigation/NavigateResponseConverter.java

```diff
@@ -136,44 +137,152 @@ private static void putLegInformation(ObjectNode legJson, ResponsePath path, int
         legJson.put("distance", Helper.round(distance, 1));
     }
 
-    private static ObjectNode putInstruction(PointList points, InstructionList instructions, int instructionIndex, Locale locale,
-                                             TranslationMap translationMap, ObjectNode instructionJson, boolean isFirstInstructionOfLeg,
-                                             DistanceConfig distanceConfig, List<PathDetail> intersectionDetails, int pointIndexFrom,
-                                             int pointIndexTo) {
-        Instruction instruction = instructions.get(instructionIndex);
-        ArrayNode intersections = instructionJson.putArray("intersections");
+    private static  List<PathDetail> filterIntersectionDetails(PointList points, List<PathDetail> intersectionDetails, int pointIndexFrom,
```

Comment:
we should have a comment saying what exactly is getting filtered (duplicated intersections)

the "why" can be in a PR description, since it might get lengthy